### PR TITLE
Category sort by name and sort by products_count

### PR DESCRIPTION
**Pull request by at-trangle**
## Description
* Add sort category:
  * Sort by name
  * Sort by products_count
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- config/define.php

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

> explain select `categories`.*, (select count(*) from `products` where `categories`.`id` = `products`.`category_id` and `products`.`deleted_at` is null) as `products_count` from `categories` where `categories`.`deleted_at` is null order by `name` desc limit 10 offset 0\G;

```
*************************** 1. row ***************************
           id: 1
  select_type: PRIMARY
        table: categories
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 52
     filtered: 10.00
        Extra: Using where; Using filesort
*************************** 2. row ***************************
           id: 2
  select_type: DEPENDENT SUBQUERY
        table: products
   partitions: NULL
         type: ref
possible_keys: products_category_id_foreign
          key: products_category_id_foreign
      key_len: 4
          ref: product_tiki.categories.id
         rows: 1
     filtered: 10.00
        Extra: Using where
2 rows in set, 2 warnings (0.00 sec)
```

> explain select `categories`.*, (select count(*) from `products` where `categories`.`id` = `products`.`category_id` and `products`.`deleted_at` is null) as `products_count` from `categories` where `categories`.`deleted_at` is null order by `products_count` desc limit 10 offset 0\G;

```
*************************** 1. row ***************************
           id: 1
  select_type: PRIMARY
        table: categories
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 52
     filtered: 10.00
        Extra: Using where; Using temporary; Using filesort
*************************** 2. row ***************************
           id: 2
  select_type: DEPENDENT SUBQUERY
        table: products
   partitions: NULL
         type: ref
possible_keys: products_category_id_foreign
          key: products_category_id_foreign
      key_len: 4
          ref: product_tiki.categories.id
         rows: 1
     filtered: 10.00
        Extra: Using where
2 rows in set, 2 warnings (0.00 sec)
```

### CODE REVIEW

File: resources/lang/en/category.php [hunk 1]
```diff
@@ -15,6 +15,10 @@
             'action' => 'Action',
             'child_category' => 'Child Category',
             'sum_product' => 'Sum Product',
+            'sort_by_name' => 'name',
+            'sort_by_products_count' => 'products_count',
+            'dir_asc' => 'ASC',
+            'dir_desc' => 'DESC',
```
Comment (at-phuongvu) [inline resources/lang/en/category.php]: cái này ở `config.define` có 1 cái global rồi định nghĩa lại làm gì ?

File: app/Http/Controllers/Admin/CategoryController.php [hunk 2]
```diff
@@ -13,11 +13,18 @@ class CategoryController extends Controller
...
+     * @param \Illuminate\Http\Request $request request
+     *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $listCategories = Category::with('parent')->withCount('products')->paginate(config('define.category.limit_rows'));
+        $listCategories = Category::with('parent')
+            ->withCount('products')
```
Comment (at-phuongvu) [inline app/Http/Controllers/Admin/CategoryController.php]: không dài quá thì cứ viết cùng hàng cũng được chứ có sao đâu ...

Comment (at-trangle) [reply app/Http/Controllers/Admin/CategoryController.php]: tại để dài quá khó nhìn nên cho xuống dễ nhìn

Comment (at-phuongvu) [reply app/Http/Controllers/Admin/CategoryController.php]: uhm huh

**Pull request merged**